### DISCUSSION TITLE
Implement `quoteProps:consistent` from Prettier

### DISCUSSION THREAD

**Discussion by uncenter** [biomejs/biome#3313] [Ideas]
I brought this up in the Discord server around a month ago though I'd like to bring it here for a more formal discussion on the merits of such a rule. The closest I got to an answer was the following from @Conaclos:

> When we brought the discussion around `quoteProps:consistent`, we noticed that most people - including those setting the option - didn't understand well the effect of the option. This current thread is another example. Most people think that `consistent` means requiring quotes for every property if at least one property is quoted. Unfortunately this is not the case. It means requiring quotes for every property if at least one property **requires** quotes. Because of the cognitive complexity of the option and it is doubtful interest we decided to not implement the option.

So it seems to be an issue with the *naming*, not the concept, of the rule. I'm wondering if seeing as that is the case renaming this option for the rule could be a viable strategy? E.g. `all-or-none` (all use quotes or none use quotes), `all-if-some` (all use quotes if some initially use quotes), `all-if-present` (all use quotes if quotes are present initially)...

I'd really love to see this change in Biome, and I'm sure I'm not the only one: according to https://github.com/search?q=%22quoteProps%22:+%22consistent%22&type=code & https://github.com/search?q=quoteProps:+%22consistent%22&type=code close to fifteen thousand configuration files are using this behavior.

**Comment by ematipico** [comment]
> So it seems to be an issue with the naming, not the concept, of the rule.

It is about naming, and it's not only that. It's also about the impact that it can have on diffs.

For example, you have a module of 1k LOC, where no properties were required.

Then, you make a PR that adds a property that **requires** quotes, and because of that all properties across 1k LOC change. You'll have formatting changes scattered, plus your changes. This isn't what a formatter should do, so I don't see enough value to have this option.
